.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,208 @@
+    * If you need to give additional context to your test please do add a `README` in this directory!
+2. In this new directory you'll need to create 4 files: `har.json`, `openapi.json`, `mock.json`, and `output.js`.
+
+#### har.json
+
+The way that `httpsnippet-client-api` works is that it takes a HAR object and constructs a snippet according to it, and what the matching operation in an OpenAPI definition, say the API request should function.
+
+This file should contain a fully defined [HAR object](http://www.softwareishard.com/blog/har-12-spec/) for your API request.
+
+If you need to use dynamic data (like base64 encoding an API key) then this file can be renamed to `har.js` and the HAR can be exported with `module.exports`.
+
+#### openapi.json
+
+Within `openapi.json` you'll want to spec out the OpenAPI definition that will be powering this request. This definition should be fully valid to the spec.
+
+If you'd like to reference one of the definitions that we provide in the [@readme/oas-examples](https://npm.im/@readme/oas-examples) package then you can rename `openapi.json` to `openapi.js` and have a single line of the following and the test suite will pick it up!
+
+```js
+module.exports = require('@readme/oas-examples/path/to/spec');
+````
+
+Additionally if you need help verifying that your OpenAPI definition is valid you can run `npx rdme validate path/to/your/definition.json`.
+
+#### mock.json
+
+This file is what is going to tell our test suite what endpoints your API test will be accessing so we can mock out those requests for you. Behind the scenes we use the amazing [nock](https://npm.im/nock) library for this. Generating this data is a bit difficult, but you can use `nock` itself to guide you along.
+
+Say you have the following snippet:
+
+```js
+const sdk = require('api')('https://raw.githubusercontent.com/readmeio/api/main/packages/httpsnippet-client-api/test/__datasets__/query/openapi.json');
+
+sdk.get('/anything', {foo: ['bar', 'baz'], baz: 'abc', key: 'value'})
+  .then(res => console.log(res))
+  .catch(err => console.error(err));
+```
+
+Running this normally you'll receive a JSON object back from the https://httpbin.org server referenced in that OpenAPI definition:
+
+```js
+{
+  args: { baz: 'abc', foo: [ 'bar', 'baz' ], key: 'value' },
+  data: '',
+  files: {},
+  form: {},
+  headers: {
+    Accept: '*/*',
+    'Accept-Encoding': 'gzip,deflate',
+    Host: 'httpbin.org',
+    'User-Agent': 'api (node)/4.2.0',
+    'X-Amzn-Trace-Id': 'REDACTED'
+  },
+  json: null,
+  method: 'GET',
+  origin: 'REDACTED',
+  url: 'https://httpbin.org/anything?key=value&foo=bar&foo=baz&baz=abc'
+}
+```
+
+To have `nock` interpret this request into data that we can use to mock out requests back with `nock` you can add modify your snippet as such:
+
+```js
+const nock = require('nock');
+
+nock.recorder.rec({
+  output_objects: true,
```

Comment:
Interesting, I've never used this `output_objects` option before! I really dig this approach, makes writing tests a loooot quicker